lib/core/src/sync/mod.rs

```diff
@@ -414,33 +418,45 @@ impl SyncService {
             }
         }
 
-        // Step 4: Split each record into two categories: swap and non-swap
-        let (decrypted_swap_info, decrypted_non_swap_info): (
-            Vec<DecryptionInfo>,
-            Vec<DecryptionInfo>,
-        ) = decrypted
+        // Step 4: Split each record into swap, last derivation index and others
+        let (decrypted_swap_info, decrypted_non_swap_info): (Vec<_>, Vec<_>) = decrypted
             .into_iter()
             .partition(|result| result.record.data.is_swap());
 
-        // Step 5: Recover the swap records' data from onchain, and commit it
+        let (decrypted_last_derivation_index_info, decrypted_others_info): (Vec<_>, Vec<_>) =
+            decrypted_non_swap_info
+                .into_iter()
+                .partition(|result| matches!(result.record.data, SyncData::LastDerivationIndex(_)));
+
+        // Step 5: Apply derivation index updates and sync wallet (step 6 requires an updated wallet)
+        for decryption_info in decrypted_last_derivation_index_info {
+            if let Err(e) = self.commit_record(&decryption_info, None) {
+                warn!("Could not commit last derivation index record: {e:?}");
+                continue;
+            }
+            succeded.push(decryption_info.record.id);
+        }
+        self.onchain_wallet.full_scan().await?;
```

Comment:
Maybe check there is a derivation index update before triggering, otherwise we do it every pull. Could also check there are swaps to recover, else there is no point scanning

Reply:
Good point. Done in [932bf1f](https://github.com/breez/breez-sdk-liquid/pull/775/commits/932bf1fb2875a9d0894c66daf547904f3c587bc2)

Reply:
But this still doesn't prevent the initial pull on a fresh instance from being significantly delayed. But I'm not sure we can get around doing the full scan, at least for as long as we continue to depend on the wallet tx map.

Reply:
This only effects an edge case though right, where the first batch of 20 addresses are unused?

I imagine if there is any large gap in the addresses and the device comes online after a while, there will be a delay until we sync the up-to-date derivation index

Reply:
I see now that I my comment above should have been located here.
I meant that if we are able to compare the derivation index to the one already scanned by the wallet at this point then we can trigger a full scan and remove the added full scan during sync.

Reply:
Thanks @roeierez. Indeed having the scan in `recover_from_onchain` prevents an unnecessary additional scan. Refactored in [0687df9](https://github.com/breez/breez-sdk-liquid/pull/775/commits/0687df9acc19c44eba588cce64e409f632e32f51)